### DISCUSSION TITLE
How to disable chromedriver download and just use devtools protocol in WDIO8?

### DISCUSSION THREAD

**Discussion by rturner** [webdriverio/webdriverio#11579] [Q&A] [answered]
Our company firewall blocks the chromedriver download.  In WDIO7 we would run npm install --chromedriver-skip-install and that worked perfectly using devtools protocol.  This flag is not respected in WDIO8.  In the config I set the automationProtocol: devtools, but on run it still tries to download chromedriver even though it won't be used.  What's the proper setup for this in 8?

I am not invoking any services in the config nor do I have any chrome/chromedriver related entries in package.json.

**Comment by christian-bromann** [comment, marked as answer]
We discourage to use `devtools` for automation since it is not behaving 100% like WebDriver and we likely won't maintain in further down the line. We recommend to have Chromedriver downloaded somewhere that matches with your Chrome and provide a path to Chrome and Chromedriver. In these cases no downloads will be initiated.

**Reply by rturner** [reply]
Thanks for the reply.  Are these the correct capabilities?  When I supply like this in the config it still attempts to download Chrome and Chromedriver at runtime.

   capabilities: [{
      browserName: 'chrome',
      'wdio:chromedriverOptions': {
        binary: '/Users/xxxxx/chrome/chromedriver-mac-x64/chromedriver'
      },
      "goog:chromeOptions": {
        binary: '/Users/xxxxx/chrome/chrome-mac-x64/GoogleChromeforTesting',
        args: [
          '--ignore-certificate-errors'
        ]
      }
    }],

**Reply by christian-bromann** [reply]
Yes, this is correct. I tested it myself and are able to run tests without an internet connection.

**Reply by rturner** [reply]
Could you try also deleting your cached chromedriver/chrome?  When that is missing I find it is trying to go download even with binary set.

**Reply by rturner** [reply]
Disregard, on updating to 8.21.0 it is now respecting the binary.  Was running on 8.13.5 and it was not.
